### ISSUE TITLE
[Q] Download an artifact without the API

### ISSUE THREAD

**Issue by quentinblampey** [wandb/wandb#7841]
Hello,

I have a simple question, but I couldn't figure out a simple way to do that.

I want users to be able to download artifacts without installing wandb. In particular, I want to implement a function that will download a file `model.ckpt` given a certain artifact name, without using `wandb.Api().artifact`. There is a public link on the UI, but it seems to use some hashing internally, and therefore I can't guess the URL of this link using the artifact name.

**TL;DR**: how to get the URL to download a file of an artifact, knowing only the artifact name?

Thanks!

**Comment by quentinblampey** [comment]
Hi @fmamberti-wandb!

I'm saving pytorch-lightning checkpoints as artifacts, which is super convenient to load for me (I download the checkpoint using the `wandb` API and then use the `load_from_checkpoint` method from pytorch-lightning to load the model).

But, ideally, I would prefer to keep wandb as an extra dependency, because users of my package will only need to load pretrained models, without re-training. In brief, I would use wandb for model training, and the package users would simply download the `model.ckpt` file and use the rest of the package without wandb.

I can of course add wandb as a main dependency, but since it's only one file that need to be downloaded, I think it could be avoided.

This is an example of a file I want to load automatically:
<img width="856" alt="Screenshot 2024-06-22 at 14 57 22" src="https://github.com/wandb/wandb/assets/33903498/612062d4-b15b-4a94-bf76-05eee9bb56a5">

**Comment by fmamberti-wandb** [comment]
Thanks for the additional details @quentinblampey. I've now raised this as a feature request for our product team to review.

**Comment by quentinblampey** [comment]
Thanks @fmamberti-wandb 👍

**Comment by quentinblampey** [comment]
Hello @fmamberti-wandb, just for your information, I'm now using `huggingface_hub` to share the checkpoints
This way I can keep wandb as an optional dependency

I'm not closing it, in case you think it's still something you want to work on, but else you can close it!
